docs/bugpattern/CheckedExceptionNotThrown.md

```diff
@@ -18,6 +16,8 @@ Response handle(Request request) {
 }
 ```
 
-Including unthrown exceptions can be reasonable where the method is overridable,
-as overriding methods will not be able to declare that they throw any exceptions
-not included.
+Unfortunately this can lead to call sites being forced to explicitly handle or
+propagate exceptions which provably can never occur. It may also lead readers of
+the code to search for possibly throwing paths where none exist.
+
+The rule checks only whether the method is overridable, not whether it is overridden.
```

Comment:
It would probably be enough, IMO, to add to the previous wording: "; this check will thus only flag non-overridable methods." (or "not flag overridable methods")

Reply:
I wanted to answer the question why Java allows redundant throws in the opening statement and not leave it for later. Currently the two statements stand in isolation, yet this is the crux of how the issue comes up in the first place.

As I look at the example now I see it showing the undesirable propagation but not the main issue: checkArgument doesn't throw IOException. I'll try come up with a better example

Reply:
@tbroyer do you understand this "It may also lead readers of the code to search for possibly throwing paths where none exist." ?